### PR TITLE
bpo-39983: Add test.support.print_warning()

### PR DESCRIPTION
**Pull request by vstinner**
Log "Warning -- ..." test warnings into `sys.__stderr__` rather than
sys.stderr, to ensure to display them even if sys.stderr is captured.

`test.libregrtest.utils.print_warning()` now calls
`test.support.print_warning()`.

https://bugs.python.org/issue39983

### CODE REVIEW

**Review by corona10 — COMMENTED** [review]
Hmm since test_ prefix means that test something in Python,
I 'd like to suggest changing name such as just `warning` (support.warning)

What do you think?

**Comment by vstinner** [comment]
> What do you think?

I renamed the function to print_warning(). The function does what its name says: it prints a warning.

File: Lib/test/libregrtest/runtest.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def cleanup_test_droppings(test_name, verbose):
                                f"directory nor file")
 
         if verbose:
-            print_warning("%r left behind %s %r" % (test_name, kind, name))
+            print_warning(f"{test_name} left behind {kind} {name!r}")
```
Comment (corona10) [inline Lib/test/libregrtest/runtest.py:330]:
```suggestion
            print_warning(f"{test_name!r} left behind {kind} {name!r}")
```

The original code looks like this, is this change intended?

**Review by corona10 — APPROVED** [review]
LGTM with nit comment.

File: Lib/test/libregrtest/runtest.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/libregrtest/runtest.py:330]: I think that it was a bug in the old code: see the print_warning() below (line 339).

Comment (corona10) [reply Lib/test/libregrtest/runtest.py:330]: Got it :)

**Comment by vstinner** [comment]
@corona10: You was right that "test_warning()" wasn't the best name ;-) print_warning() is better.
